mempool/src/shared_mempool/types.rs

```diff
@@ -250,13 +252,77 @@ impl Ord for BatchId {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct TimelineId(pub Vec<u64>);
+
+impl TimelineId {
+    pub(crate) fn new(num_buckets: usize) -> Self {
+        Self(vec![0; num_buckets])
+    }
+
+    pub(crate) fn update(&mut self, batch_id: &MultiBatchId) {
+        if self.0.len() != batch_id.0.len() {
+            return;
+        }
+
+        let updated: Vec<_> = self
+            .0
+            .iter()
+            .zip(batch_id.0.iter())
+            .map(|(&cur, &(_start, end))| std::cmp::max(cur, end))
```

Comment:
should we check that start matches?

i.e that cur == start, and then that set to end? 

or can this sometimes be misaligned, such that cur might be > end?

Reply:
This can be misaligned, because older broadcasts that needed retries can be sent to `update`.